wordpress-mobile/WordPress-Android
# IA Feature Flag remove for me move and create fab v1

### Note that this PR is based and needs #11064 and #11069. So those 2 should be first merged in develop before to merge this one.

This PR has been created to remove the IA feature flag (`INFORMATION_ARCHITECTURE_AVAILABLE`) for the features listed below so to have them released in vanilla develop for beta testing/release.

# What needs to be checked

## 1. Access to the Me screen was moved

<p align="center">
  <img width="300" src=https://user-images.githubusercontent.com/47797566/72226537-6ab58600-3592-11ea-944b-90b34951c386.png>
</p>

We removed the Me screen access from the bottom navigation bar and we moved it to be a menu icon in the My Site screen.

### To test
- Compile and run the vanilla flavor
- Check that the Me menu item is not in the bottom nav bar
- Check that the me options menu icon is only available in the My site section
- Tap on the me options menu icon to check that the Me section is visualized
- Press the back arrow and check you are brought again in the My Site screen

## 2. Possibility to create a new post or page from My Site

| ![image](https://user-images.githubusercontent.com/47797566/72226673-04c9fe00-3594-11ea-8c07-8f50d4474bbf.png) | ![image](https://user-images.githubusercontent.com/47797566/72226702-3a6ee700-3594-11ea-908d-a3a8a69aac05.png) |
|---|---|

We removed the New Post menu item from the bottom navigation bar in favor of a FAB button in My Site. The FAB button opens a bottom sheet from which it is possible to create a new Blog Post and also a new Site Page. The feature is announced by a tooltip according the logic reported in the test steps below.

### To test
- Check that the Me and the New Post menu items are not in the bottom nav bar
- Switch between My site/Reader/Notifications pages and check that the FAB/Tooltip is only available in the My site screen
- Rotate the device and check the FAB/Tooltip is correctly presented
- Set a RTL language like Hebrew and check the FAB/Tooltip is correctly presented, then restore the English language
- Use the app without tapping on the FAB (for example create a post from Blog Posts list) and check that the tooltip is always presented when you come back to the My Site page also from Reader or Notifications
- Tap on the tooltip to get it hide. Check that the tooltip is presented again when you come back to the My Site page from Reader or Notifications
- Use the FAB and check that after that the tooltip is no more visualized. Also close and open the app to confirm.
- Select the FAB and check that a bottom sheet is opened (also note that the tooltip is hidden. Once tapped on the FAB the tooltip is not shown any more and there is no way to have it back unless the app storage data is cleared or the user fully uninstall and reinstall the app). 
- Check that the Editor is opened when selecting the new Blog post/Site page from the bottom sheet
- Create a post/page and tap on the back arrow. You should be back on the main screen and notified by a snackbar that the post was saved on line.
- Access to the Pages/Posts list and check the new created Page/Post was created and available in the correct list (Published/Draft).

## 3. QuickStart reviewed based on new IA architecture (some more info in #11042 if needed)
In the IA project we removed the add post button in the main navigation bottom bar. As a result of this, the QuickStart tutorials are still ok, but the PUBLISH_POST_TUTORIAL that was targeting the removed bottom button is no more ok. We aligned the `PUBLISH_POST_TUTORIAL` to the `CREATE_NEW_PAGE` behavior. That is targeting the `row_blog_posts` in the My Site list of sections (in analogy to the `row_pages` targeted by the `CREATE_NEW_PAGE` tutorial)

### To test
- Create a new site from the app and Accept on the dialog below to get access to the QuickStart tutorials 
<p align="center">
  <img width="200" src=https://user-images.githubusercontent.com/47797566/72227048-56748780-3598-11ea-84c3-4687add31f24.png>
</p>

- go through the tutorials and check that nothing changed for the QuickStart except the Publish Post for which check the next step
- for the Publish Post you need to get a similar behavior as for the create page tutorial but targeting the blog posts row in My Site. Also some minor changes in the copy of the snackbars are made to account for the Posts (not Pages) scenario naming.

## 4. Reader + Notifications unchanged
The IA Reader modifications are still behind the `INFORMATION_ARCHITECTURE_AVAILABLE` feature flag that was released in the develop branch in 13.8. Since we did not touch the related code we should expect no difference in behavior.

### To test
To complete this check smoke test the `Reader` and `Notifications` sections.

## Notes
While working on this PR we found the following:
1. UX issue that this PR fixes. In `launchZendeskMyTickets` the previous code version was doing a `mBottomNav.setCurrentSelectedPage(PageType.ME)` so that in case of a zendesk support push notification tapped by the user, when user comes back from the HelpActivity user can found the Me page selected. Now we removed the ME page from the bottom sheet so we made here a `mBottomNav.setCurrentSelectedPage(PageType.MY_SITE)` and we launch the `HelpActivity` by a newly added funtion (`viewHelpAndSupportInNewStack`) that populates the back stack correctly. In this way the UX should be fixed. To test this one:
     - Create a support ticket from the app
     - Trigger a notification from zendesk answering to the above ticket (@malinajirka I have all the zendesk setup in place to trigger this so in case you need to setup it just ping me and i can trigger this for you while you reviewing 😊)
     - Tap on the zendesk notification and check that you get the `My tickets` opened.
     - Go back from this screen and check you are presented with the correct screens sequence from the backstack (namely `Help and Support`, `Me`, `My Site`)
2. Fix bug to avoid showing the FAB if user has no sites (see here 3d1836ab2ade962070fb49c2e1fe7fcf780cc4ab)
3. Some modifications to connected tests for Login flow were needed since the nav_me button is no more on the bottom nav bar (see 2d8b720eb0054f4cffbf6dc44cac9f74d2c2038c)

PR submission checklist:
